File: project/backend/springboot/src/core/src/main/java/mvm/micro/core/infrastructure/controller/HealthCheckController.java
package mvm.micro.core.infrastructure.controller;

import mvm.micro.core.application.usecase.HealthCheck.HealthCheckQuery;
import mvm.micro.shared.infrastructure.BaseResponse;
import mvm.micro.shared.infrastructure.controller.ApiController;
import mvm.micro.shared.domain.bus.command.CommandBus;
import mvm.micro.shared.domain.bus.query.QueryBus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/health")
public class HealthCheckController extends ApiController {

    public HealthCheckController(CommandBus commandBus, QueryBus queryBus) {
        super(commandBus, queryBus);
    }

    @GetMapping
    public ResponseEntity<Object> healthCheck() {
        try {
            BaseResponse response = ask(new HealthCheckQuery());
            return ResponseEntity.ok(response.getData());
        } catch (Exception e) {
            return ResponseEntity.status(500).body(e.getMessage());
        }
    }

    @Override
    protected void registerExceptions() {
        // Registro de excepciones similar a Symfony
    }
}

File: project/backend/django/myproject/core/infrastructure/controllers/health/health_check_controller.py
from django.http import JsonResponse
from myproject.core.application.UseCase.HealthCheck.health_check_query import HealthCheckQuery
from myproject.shared.infrastructure.bus.query_bus import get_query_bus
from myproject.shared.infrastructure.controller.api_controller import ApiController


class HealthCheckController(ApiController):
    def __init__(self):
        qb = get_query_bus()
        super().__init__(query_bus=qb)

    def get(self, request):
        query = HealthCheckQuery()
        response = self.ask(query)
        return JsonResponse(response, safe=False)

    def register_exceptions(self) -> dict:
        return {
            ValueError: 400,
        }
